service-python/assessclaimdc7101/src/lib/bp_calculator.py

```diff
@@ -21,6 +21,30 @@ def sort_bp(bp_readings):
     return bp_readings
 
 
+def deduplicate(bp_readings):
+    """
+    Return bp readings with any lighthouse duplicates removed. A duplicate is identified by having the same diastolic
+    value, systolic value and date. HDR data has organization set to VAMC Other Output Reports.
+    :param bp_readings: full list of BP readings
+    :return: deduplicated list
+    """
+    deduplicated_readings = []
+    for reading in bp_readings:
+        duplicate = False
+        if reading["dataSource"] == "LH":
+            for bp_comp in bp_readings:
+                if reading["diastolic"]["value"] == bp_comp["diastolic"]["value"] \
+                    and reading["systolic"]["value"] == bp_comp["systolic"]["value"] \
+                        and reading["date"] == bp_comp["date"] \
+                        and bp_comp["organization"] == "VAMC Other Output Reports":
+                    duplicate = True
+                    bp_readings.remove(reading)
```

Comment:
removing from bp_readings while inside a for loop for bp_readings (and double on that) is a bad practice that hurt me once or twice. I do not know what happens in python actually. I suggest you reorganize.

Reply:
> Where exactly is the unit test for this?

I added some unit tests in most recent commits

Reply:
I will removed this line